src/core/interaction_data.cpp

```diff
@@ -162,6 +163,7 @@ void ia_params_set_state(std::string const &state) {
   boost::archive::binary_iarchive ia(ss);
   ia_params.clear();
   ia >> ia_params;
+  ia >> max_seen_particle_type;
```

Comment:
I think you have to set max_seen_particle_type to the maximum of its current value and the value from the archive, as it's current value could be higher than the stored one.

Reply:
but then I'm also not allowed to clear ```ia_params```, somebody could have created interactions before loading the checkpoint

Reply:
in my opinion the user cannot expect that espresso is able to merge current state with saved state